src/main/java/strategy/activities/ActivitiesHandler.java

```diff
@@ -0,0 +1,5 @@
+package strategy.activities;
+
+public interface ActivitiesHandler {
```

Comment:
Considering my above comments, your activity (operation) handlers could receive FruitTransaction entity and encapsulate all the transaction-related processing